packages/lib/models/Folder.test.ts

```diff
@@ -223,6 +223,34 @@ describe('models/Folder', () => {
 		expect(sortedFolderTree[2].id).toBe(f6.id);
 	}));
 
+	it('should sort folders with special chars alphabetically', (async () => {
+		const unsortedFolderTitles = ['ç', 'd', 'c', 'Ä', 'b', 'a'].map(firstChar => `${firstChar} folder`);
+		for (const folderTitle of unsortedFolderTitles) {
+			await Folder.save({ title: folderTitle });
+		}
+
+		const folders = await Folder.allAsTree();
+		const sortedFolderTree = await Folder.sortFolderTree(folders);
+
+		// same set of titles, but in alphabetical order
+		const sortedFolderTitles = ['a', 'Ä', 'b', 'c', 'ç', 'd'].map(firstChar => `${firstChar} folder`);
+		expect(sortedFolderTree.map(f => f.title)).toEqual(sortedFolderTitles);
+	}));
+
+	it('should sort numbers ascending', (async () => {
+		const unsortedFolderTitles = ['10', '1', '2'].map(firstChar => `${firstChar} folder`);
+		for (const folderTitle of unsortedFolderTitles) {
+			await Folder.save({ title: folderTitle });
+		}
+
+		const folders = await Folder.allAsTree();
+		const sortedFolderTree = await Folder.sortFolderTree(folders);
```

Comment:
Regarding the failed test:

1. I guess we need to pass the sorting options, like `Folder.allAsTree(null, { order: [{ by: 'title', dir: 'ASC' }]})`.
2. `sortFolderTree` seems to sort even once more and uses the wrong sorting for numbers. I think it's only used when moving notes to another notebook:

![grafik](https://github.com/laurent22/joplin/assets/33229141/71c0ec3c-40b6-49d8-94bc-9de0b96e2970)

Reply:
Hey marph91, I have made the changes, the test is passing now. 

1) I have changed the sensitivity in collator to accent.
2) I have used the same approach that I did everywhere else instead of `localCompare`.